exercise_10/transformCollectionToString.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Write JS function "transformCollectionToString" to transform collection into string
+    - if array has elements, then return array as string, joined by '|' (pipe character)
+    - otherwise, return an empty string
+ * input: array (collection) 
+ * output: string
+ */
+
+/**
+ * CASE #1
+ * input: ['apple', 'grapes', 'strawberries']
+ * output: "apple | grapes | strawberries"
+ */
+
+/**
+ * CASE #2
+ * input: []
+ * output: ""
+ */
+
+/**
+ * transformCollectionToString
+ * @param {Array} collection
+ * @returns {string}
+ */
+
+const transformCollectionToString = function (collection) {
+  if (Array.isArray(collection)) {
```

Comment:
..._if array has elements_..., not if _collection is an array_.

We defined a method to check whether a collection has elements or not.

FIX: use abstractions, not implementations.